### ISSUE TITLE
Rule configuration request: Ignore based on annotations

### ISSUE THREAD

**Issue by ZacSweers** [arturbosch/detekt#2563]
## Expected Behavior of the rule
```yml
LongParameterList:
  active: true
  functionThreshold: 7
  constructorThreshold: 7
  ignoreDefaultParameters: false
  ignoreDataClasses: true
  ignoreAnnotated: "javax.inject.Inject,javax.annotation.Generated"
```

## Context
Currently the `LongParameterList` rule applies to all constructors but has an opt-out for data classes. I'd like to suggest adding a new configuration to ignore based on annotations, such as `javax.inject.Inject`. This comes up often in our DI plumbing

**Comment by schalkms** [comment]
Please go ahead. I think this a good idea.
Do you have resources to tackle this? @ZacSweers 
It should be very easy to add this in the `LongParameterList` class in the following function.
The function can be extended to check by `constructor.annotations.any { ignoreAnnotated.contains(it.text) }`

https://github.com/arturbosch/detekt/blob/91fdc64fecb858045059bf544fa9bfb6017c8c1d/detekt-rules/src/main/kotlin/io/gitlab/arturbosch/detekt/rules/complexity/LongParameterList.kt#L65-L71

**Comment by ZacSweers** [comment]
Done: #2570
